src/dmd/e2ir.d

```diff
@@ -2964,6 +2908,19 @@ extern (C++) class ToElemVisitor : Visitor
 
     override void visit(CondExp ce)
     {
+        if (auto ve = ce.econd.isVarExp)
+            if (ve.var.ident == Id.ctfe)
+            {
+                elem *e = toElem(ce.e2, irs);
+                if (irs.params.cov && ce.e2.loc.linnum)
+                    e = el_combine(incUsageElem(irs, ce.e2.loc), e);
+                if (tybasic(e.Ety) == TYstruct)
+                    e.ET = Type_toCtype(ce.e2.type);
+                elem_setLoc(e, ce.loc);
+                result = e;
+                return;
+            }
```

Comment:
This doesn't look relevant to the change, and the backend should already optimize this case.

Reply:
If I don't do this here, the `true` branch of the cond reaches `visit(CatExp)`, which is triggers these `assert`s [1] [2]. In addition, it doesn't make sense to lowert the `CatExp` twice: in expressionsem.d and here.

[1] https://github.com/dlang/dmd/blob/3f1b2c4e2d7b3a717e8e0158cd4b6731b04e28ad/src/dmd/e2ir.d#L2717
[2] https://github.com/dlang/dmd/blob/3f1b2c4e2d7b3a717e8e0158cd4b6731b04e28ad/src/dmd/e2ir.d#L2722